tests/unit/oidc/models/test_activestate.py

```diff
@@ -0,0 +1,366 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pretend
+import pytest
+
+from tests.common.db.oidc import (
+    ActiveStatePublisherFactory,
+    PendingActiveStatePublisherFactory,
+)
+from warehouse.oidc.interfaces import SignedClaims
+from warehouse.oidc.models import _core
+from warehouse.oidc.models.activestate import (
+    ActiveStatePublisher,
+    PendingActiveStatePublisher,
+)
+
+ORG_ID = "00000000-0000-1000-8000-000000000000"
+PROJECT_ID = "00000000-0000-1000-8000-000000000001"
+USER_ID = "00000000-0000-1000-8000-000000000002"
+BRANCH_ID = "00000000-0000-1000-8000-000000000003"
+
+
+def test_lookup_strategies():
+    assert (
+        len(ActiveStatePublisher.__lookup_strategies__)
+        == len(PendingActiveStatePublisher.__lookup_strategies__)
+        == 1
+    )
+
+
+def new_signed_claims(
+    sub: str = "fake:fake:fake:fake",
+    organization_id: str = ORG_ID,
+    org_url_name: str = "fakeorg",
+    project_id: str = PROJECT_ID,
+    project_name: str = "fakeproject",
+    project_path: str = "fakeorg/fakeproject",
+    user_id: str = USER_ID,
+    project_visibility: str = "public",
+    branch_id: str | None = None,
+) -> SignedClaims:
+    project_name = "fakeproject"
+    org_url_name = "fakeorg"
+    claims = SignedClaims(
+        {
+            "sub": sub,
+            "organization_id": organization_id,
+            "organization_url_name": org_url_name,
+            "project_id": project_id,
+            "project_name": project_name,
+            "project_path": project_path,
+            "user_id": user_id,
+            "project_visibility": project_visibility,
+        }
+    )
+    if branch_id:
+        claims["branch_id"] = branch_id
+    return claims
+
+
+class TestActiveStatePublisher:
+    def test_publisher_name(self):
+        publisher = ActiveStatePublisher()
+
+        assert publisher.publisher_name == "ActiveState"
+
+    def test_publisher_url(self):
+        publisher = ActiveStatePublisher()
+
+        assert publisher.publisher_url == "https://platform.activestate.com/oauth/oidc"
```

Comment:
I'm not clear on what this field should be.  Is this meant to match the ISSUER?  Or can this be anything we like?  If it's the latter, I'll change this to `https://platform.activestate.com/`.  If it's the former, it would be more clear field had a different name, indicative of it's meaning.

Reply:
This is not the issuer, it's a URL that uniquely identifies the publisher. E.g. for GitHub, this is a link to a repository, possibly also including a specific commit SHA if one is available for the identity used: https://github.com/pypi/warehouse/blob/5a0289e2d9df46a1e3d823a6c00b9aa74bd03417/warehouse/oidc/models/github.py#L201-L207

Reply:
Ahhhh, so this is a "publisher" as in the user that owns the code is publishing it to PyPI, not the OIDC publisher/provider.  Ok.  I know what to put here now.